lib/src/builder.dart

```diff
@@ -189,6 +136,80 @@ Stream<GeneratedOutput> _processUnitMember(
   }
 }
 
+/// Wraps multiple [Generator]s and exposes them as a [Builder] which produces
+/// Dart `part` files.
+///
+/// ```dart
+/// // Creates a new builder that runs the following two generators.
+/// new PartBuilder([
+///   new JsonSerializableGenerator(),
+///   new JsonLiteralGenerator(),
+/// ]);
+/// ```
+class PartBuilder extends _Builder {
+  final bool _requireLibraryDirective;
+
+  /// Wrap [generators] in a [Builder]-compatible API.
+  ///
+  /// May set [requireLibraryDirective] to `false` in order to opt-in to
+  /// supporting a `1.25.0` feature of `part of` being usable without an
+  /// explicit `library` directive. Developers should restrict their `pubspec`
+  /// accordingly:
+  /// ```yaml
+  /// sdk: '>=1.25.0 <2.0.0'
+  /// ```
+  PartBuilder(List<Generator> generators,
+      {OutputFormatter formatOutput,
+      String generatedExtension: '.g.dart',
+      bool requireLibraryDirective: true})
+      : _requireLibraryDirective = requireLibraryDirective,
+        super(generators,
+            generatedExtension: generatedExtension, formatOutput: formatOutput);
+
+  @override
+  void _addPreamble(LibraryElement library, BuildStep buildStep,
+      AssetId outputId, StringBuffer contentBuffer) {
+    var asset = buildStep.inputId;
+    var name = nameOfPartial(
+      library,
+      asset,
+      allowUnnamedPartials: !_requireLibraryDirective,
+    );
+    if (name == null) {
+      var suggest = suggestLibraryName(asset);
+      throw new InvalidGenerationSourceError(
+          'Could not find library identifier so a "part of" cannot be built.',
+          todo: ''
+              'Consider adding the following to your source file:\n\n'
+              'library $suggest;');
+    }
+    final part = computePartUrl(buildStep.inputId, outputId);
+    if (!library.parts.map((c) => c.uri).contains(part)) {
+      // TODO: Upgrade to error in a future breaking change?
+      log.warning('Missing "part \'$part\';".');
+    }
+    contentBuffer.writeln('part of $name;');
+    contentBuffer.writeln();
+  }
+
+  @override
+  String toString() => 'PartBuilder:$generators';
+}
+
+/// Wraps a [Generator] and exposes it as a [Builder] which produces a Dart
```

Comment:
How do you mean?

I also considered keeping a single class with two constructors, `GeneratorBuilder` and `GeneratorBuilder.standalone`. Is that what you're suggesting?